### ISSUE TITLE
Future of the code repository

### ISSUE THREAD

**Issue by TCH68k** [punesemu/puNES#91]
May i ask, what is the longterm plan about storing the project on GitHub? Recently it has begun to annoy users by asking "security codes" from them after login; forcing them to login into their email accounts, to read the sent security code. The only alternative is switching to 2FA. What if this will not be an alternative, but the only way to login? I'd rather jump in front of a locomotive, than give microsoft my phone number.

**Comment by RokkumanX** [comment]
I agree, GitHub have started to be quite an annoyance to use, I don't want to verify my account every single time I want to login.

And every once in a while when I try to login to my Gmail account to get the GitHub code, Gmail wants to be verified so I have to get that code too.

So tedious, why even have passwords when you have to verify every day.

As for the puNES code it's pretty much up to @punesemu to decide whether to keep it on GitHub or not, but again I don't get why all these verifications is necessary.
I hate it...

**Comment by TCH68k** [comment]
These verifications are exists for exactly to annoy you until you bite the bullet and choose 2FA and give these corporations - whether ms or google - your mobile number which will make them able to identify you in real life.

As for GMail, i can suggest you to change to ProtonMail. They are using GPG encryption between the mailbox and your computer and only you can open it, not even them. All of their email ecosystem is open source, so their work is transparent, you can double check their statements. The only disadvantage is the basic storage size, which is only half GB, but for free, that is not bad at all. And you can purchase a bigger box for a really marginal sum per month.

Of course, there are other alternatives as well, here are some of them: https://www.privacytools.io/providers/email/

And as for the puNES codebase, i agree, it's up to Fabio to decide.

**Comment by TCH68k** [comment]
I am sure, you've already heard about microsoft had taken down the youtube-dl project's repository from here on the demand of the RIAA. While this seems unrelated, it is not.
The project youtube-dl made a generic tool to download any kind of content from youtube, including any free video; it's purpose was not piracy, even if it could be done with it. (Although one can ask, if downloading a certain video is piracy, then why that video has not been removed from there?)

The same applies for puNES or any kind of emulator: they are generic tools, to run programs which were crafted for any particular platform, including free games, demos, tutorials or anything, not just grey-zone abandonware; their purpose is not piracy. 
However if this case will be used as precedence, then henceforth any kind of software can be accused of being a tool for "piracy" and none of them will be safe on github anymore.
Considering Nintendo already attacking retrogaming sites and forcing them to take down not just ROMs, but emulators too; puNES' repository can be a target at one point and microsoft will pull the plug, just as they did with youtube-dl.

Fabio, if you read this, i think you could reconsider moving your code and it's development to a safer place, which is not willing to surrender to these companies and will not take down your work at the first snap of their fingers...

**Comment by punesemu** [comment]
I had been thinking about it for a while and was considering possible alternatives. I was moving towards gitlab, I had also taken a look at bitbucket but the fact that it was owned by Atlassian doesn't convince me very much.

**Comment by TCH68k** [comment]
Gitlab did not worked out?

**Comment by punesemu** [comment]
I haven't tried yet, what I meant by "I was moving towards gitlab" was that I was looking into several possibilities and among them the best seems to be gitlab.

**Comment by punesemu** [comment]
For now I've created and imported the project on [GitLab](https://gitlab.com/FHorse/puNES). I need some time to switch off from GitHub because I have to understand how to integrate the same things I use here.

**Comment by TCH68k** [comment]
That's great to hear. Take your time.
